blog/2020-06-01-getCardSize.md

```diff
@@ -0,0 +1,42 @@
+---
+author: Bram Kragten
+authorURL: https://github.com/bramkragten
+authorTwitter: bramkragten
+title: getCardSize can now be async
+---
+
+Ever since we introduced lazy loading to cards, getting the card size of a lazy loaded card was hard.
+
+We used to send out an error element before the element was loaded, that would have a `getCardSize` function. But that would be the wrong size.
+When the element would be defined we would fire and rebuild event so the right card would be recreated.
+
+In 0.110 we stopped doing this, we would give back the correct element, but the element constructor would not be loaded yet, so it doesn't have the `getCardSize`.
+When the constructor is loaded the element will be upgraded, and we set the config. From that moment we can call `getCardSize`.
+
+In this version we changed the logic for `getCardSize` so it will wait for this. This means some cards, like stacks, will return a promise because they have to wait for their childeren to be defined.
+
+If you are a custopm card develeoper, and your custom card uses `getCardSize` to get the size of other cards, you have to adjust it to handle these promises.
```

Comment:
```suggestion
In this version, we changed the logic for `getCardSize` so it will wait for this. This means some cards, like stacks, will return a promise because they have to wait for their children to be defined.

If you are a custom card developer and your custom card uses `getCardSize` to get the size of other cards, you have to adjust it to handle these promises.
```